src/controllers/scenario/__tests__/index.test.js

```diff
@@ -17,30 +19,37 @@ const execute = async () => {
   }
 }
 
-describe('/scenario/{example} endpoint', () => {
-  let res
-  let stubs = {}
-
-  beforeEach(async () => {
-    stubs.getRandomProcessName = stub(util, 'getRandomProcessName')
+const setupMocks = (stubs, options = {}) => {
+  before(async () => {
+    stubs.getRandomProcessName = stub(paramUtil, 'getRandomProcessName')
     stubs.getRandomProcessName.returns('out-of-bounds-read-cheri_foo')
+    stubs.checkExecutable = stub(execUtil, 'checkExecutable')
+    stubs.checkExecutable.resolves(options.checkExecutable === undefined ? true : options.checkExecutable)
     stubs.exec = stub(child, 'exec')
-    stubs.exec.onCall(0).yields(null, 'stdout - success')
+    stubs.exec.onCall(0).yields(options.exec.err, options.exec.stdout)
     stubs.exec.onCall(1)
-    res = await execute()
   })
 
-  afterEach(() => {
+  after(() => {
     stubs.getRandomProcessName.restore()
+    stubs.checkExecutable.restore()
     stubs.exec.restore()
   })
+}
 
-  describe('if executing binaries fails', () => {
-    beforeEach(async () => {
-      stubs.exec.restore()
-      stubs.exec = stub(child, 'exec')
-      stubs.exec.onCall(0).yields({ message: 'error' }, 'stdout - some error output')
-      stubs.exec.onCall(1)
+describe('/scenario/{example} endpoint', () => {
```

Comment:
Love it! You have done exactly what I have had in my mind including unit testing.